modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/utils/iframes.js

```diff
@@ -14,43 +14,38 @@
 
 import {OPEN_MODAL, OPEN_MODAL_FROM_IFRAME} from './eventsDefinitions';
 
-const iframeHandlerModalNamespace = 'iframe-handler-modal_';
-let modalsCounter = 0;
-const registeredModals = new Map();
+export function isPageInIframe() {
+	return window.location !== window.parent.location;
+}
 
-Liferay.on(OPEN_MODAL_FROM_IFRAME, (payload) => {
-	let firstAvailableModalId = null;
+const registeredModals = new Set();
+let modalsCounter = 0;
 
-	// eslint-disable-next-line no-for-of-loops/no-for-of-loops
-	for (const [modalId, modalRef] of registeredModals.entries()) {
-		if (modalRef) {
-			firstAvailableModalId = modalId;
-			break;
+if (!isPageInIframe()) {
+	Liferay.on(OPEN_MODAL_FROM_IFRAME, (payload) => {
+		if (!registeredModals.size) {
+			throw new Error(`No registered modals founds`);
```

Comment:
If the purpose of this check is to prevent some temporary edge case, like opening modal during navigation/load, it's better to fail gracefully:
```js
if (!registeredModals.size) {
    return;
}
```

If logging error is useful to us, we should fail gracefully and [log warning in dev environment](https://github.com/liferay/liferay-frontend-projects/blob/dc7d947da927de33e556c7570c007325efa560dc/guidelines/dxp/debugging.md#use-of-node_envdevelopment). 

Lastly, if you are sure we want to throw an error here, if error is useful even to customers, a minor grammar fix: 'No registered modals found.'